titus-common/src/main/java/com/netflix/titus/common/runtime/SystemLogService.java

```diff
@@ -0,0 +1,24 @@
+package com.netflix.titus.common.runtime;
+
+/**
+ * An API for logging critical system events (errors, exceptions, invariant violations) with supplementary metadata information.
+ * The events are classified by their category and priority for better searchability.
+ */
+public interface SystemLogService {
+
+    enum Priority {
+        Low,
+        Medium,
+        High
+    }
+
+    enum Category {
+        RecoverableError,
```

Comment:
Some comments around your definitions of `recoverable` and `unrecoverable` would be helpful.  Do these refer to the ability to retry? `transient` vs `permanent`?

Reply:
Yes. We could use those terms instead.